app/controllers/concerns/authorizable.rb

```diff
@@ -13,6 +13,18 @@ def ensure_authenticated
     return render_error status: :unauthorized unless current_user
   end
 
+  # PermissionsChecker service will return a true or false depending on whether the current_user's role has the provided permission_name
+  def ensure_authorized(permission_name)
+    # TODO - should this return a status: :not_found instead of :unauthorized if the user doesn't have the permission AND the param doesn't exists?
+    return render_error status: :unauthorized unless PermissionsChecker.new(
+      current_user:,
+      permission_name:,
+      user_id: params[:id],
+      friendly_id: params[:friendly_id],
+      record_id: params[:record_id]
+    ).call
+  end
```

Comment:
The PermissionsChecker works this way:
The request either has:
1. a current_user that has the appropriate permission to act on a resource
2. the current_user is trying to act on its own resource as per the matching of current_user.resource_id == params[:ressource_id]

An awkward situation that could arrive is if the request does not have:
1. a current user with the appropriate permission
3. the provided param is not a valid user_id, friendly_id, or record_id

Reply:
I feel that :unauthorized is fine

Reply:
It's a bit more specific to what the error actually is